### PR TITLE
scorrodi/gr1: added software DR

### PR DESCRIPTION
**Pull request by corrodis**
and function the get firmware flavour

### CODE REVIEW

File: dtcInterfaceLib/CFOandDTC_Registers.cpp [hunk 1]
```diff
@@ -141,6 +141,18 @@ std::string DTCLib::CFOandDTC_Registers::ReadDesignDate(std::optional<uint32_t>
 	return o.str();
 }
 
+/// <summary>
+/// Read the DTC/CFO firmware flavour
+/// </summary>
+/// <returns>Returns the firmware flavour: C for CFO, D for DTC, E for CRVDTC</returns>
+std::string DTCLib::CFOandDTC_Registers::ReadDesigFlavour(std::optional<uint32_t> val)
```
Comment (rrivera747) [inline dtcInterfaceLib/CFOandDTC_Registers.cpp]: I think we should trade "ReadDesignFlavour" for something like "isCRVDTCDesignFlavour()," "isNonCRVDTCDesignFlavour()," and "isCFODesignFlavour()" which each just returns true/false. This way the implementation is contained within getCFOandDTCRegisters and the interpretation is simple by only using booleans.

File: dtcInterfaceLib/DTC_Registers.cpp [hunk 2]
```diff
@@ -908,6 +908,7 @@ DTCLib::RegisterFormatter DTCLib::DTC_Registers::FormatDTCControl()
 	// form.vals.push_back(std::string("Bit-25 Reset DDR Interface:             [") + (ReadResetDDR(form.value) ? "x" : " ") + "]");
 	// form.vals.push_back(std::string("Bit-24 CFO Emulator DRP Enable:         [") + (ReadCFOEmulatorDRP(form.value) ? "x" : " ") + "]");
 	form.vals.push_back(std::string("Bit-23 DTC Autogenerate DRP:            [") + (ReadAutogenDRP(form.value) ? "x" : " ") + "]");
+	form.vals.push_back(std::string("Bit-22 Software DRP:                    [") + (ReadSoftwareDRP(form.value) ? "x" : " ") + "]");
```
Comment (rrivera747) [inline dtcInterfaceLib/DTC_Registers.cpp]: Bit-22 no longer controls the Software DRP enable. Bit-23 being false enables the Software DRPs now.

File: dtcInterfaceLib/DTC_Registers.h [hunk 3]
```diff
@@ -978,6 +980,12 @@ class DTC_Registers : public CFOandDTC_Registers
 	void SetROCDCSResponseTimer(uint32_t timer);
 	RegisterFormatter FormatROCDCSResponseTimerPreset();
 
+	// Software DataRequests
+        void SetDataRequest(const DTC_EventWindowTag& ts);
```
Comment (rrivera747) [inline dtcInterfaceLib/DTC_Registers.h]: I think we should change the name pattern here from "DataRequest" to "SoftwareDataRequest" so it is clear what these functions do.

File: dtcInterfaceLib/CFOandDTC_Registers.cpp (hunk 1 above)
Comment (corrodis) [reply dtcInterfaceLib/CFOandDTC_Registers.cpp]: done

File: dtcInterfaceLib/DTC_Registers.cpp (hunk 2 above)
Comment (corrodis) [reply dtcInterfaceLib/DTC_Registers.cpp]: done

File: dtcInterfaceLib/DTC_Registers.h (hunk 3 above)
Comment (corrodis) [reply dtcInterfaceLib/DTC_Registers.h]: done

**Review by rrivera747 — APPROVED** [review]
Looks good, thanks!

**Review by eflumerf — APPROVED** [review]
There's commented code in here that may be better deleted if it's no longer relevant

**Comment by rrivera747** [comment]
> There's commented code in here that may be better deleted if it's no longer relevant

We are keeping this particular commented to code to make it clear that Software Data Requests cannot be disabled in the latest DTC versions. It is an exclusive mode with Automatic Data Requests; one mode or the other are in operation at any time.

**Pull request merged**